inst/BS5/assets/pkgdown.scss

```diff
@@ -103,6 +113,9 @@ aside {
   .list-unstyled li { margin-bottom: 0.5rem; }
   .dev-status .list-unstyled li { margin-bottom: 0.1rem; }
 }
+[data-bs-theme="dark"] aside .roles {
+  color: mix($body-color-dark, $body-bg-dark, 80%);
+}
```

Comment:
For anything that's a mix of `$body-color` and `$body-bg`, I'd recommend using transparency. For example, this color `mix($body-color, $body-bg, 80%)` is roughly equivalent to `$body-color` with 80% opacity over an element with `$body-bg` background color.

```css
aside .roles {
  color: rgba(var(--bs-body-color), 0.8);
}
```

To keep the `$pkgdown-footer-color` variable, you can move that color property value into the Sass variable

```scss
$pkgdown-footer-color: rgba(var(--bs-body-color), 0.8) !default;

aside .roles {
  color: $pkgdown-footer-color;
}
```

Another thing you can do is to reflect the Sass variable into a CSS variable and then use the CSS variable wherever needed. This is generally what we've done in bslib and shiny to bridge the gaps between Sass and CSS variables.

```scss
$pkgdown-footer-color: rgba(var(--bs-body-color), 0.8) !default;

:root {
  --pkgdown-footer-color: #{$pkgdown-footer-color};
}

aside .roles {
  color: var(--pkgdown-footer-color);
}
```

In the last case, the footer color in dark mode can be extended with user-contributed Sass like this.

```scss
$pkgdown-footer-color: black;

// or instead set the footer color with CSS variables
// (only one approach should be used)
:root {
  --pkgdown-footer-color: black;
}

@include color-mode(dark) {
  --pkgdown-footer-color: white;
}
```

Reply:
I submitted too early by accident, a few things to clarify:

1. Bootstrap has a few of these `-rgb` suffixed variables that are intended for this usage in `rgba()`. https://getbootstrap.com/docs/5.3/customize/color/#colors
2. There's a convention of using `RGBA()` in Sass when a Sass function masks a CSS function. In Sass, `rgba()` will result in a computed color value, whereas `RGBA()` is passed through to CSS.
3. When using Sass variables as the value of a custom property (css variable), you need to wrap the Sass variable in `#{}`. CSS variables are extremely permissive about their values, so this is the inlining mechanism Sass uses.

Reply:
What's the advantage of using opacity? The downsides are that you get a suboptimal mixing algorithm and you have to think more carefully what's behind the text.

Reply:
I think you're making two slightly different points. Let me try and summarise to make sure I understand correctly.

Specifically for the case of font colours (not backgrounds) that apply some overlay to body colour, opacity is a good enough colouring mixing and simplifies the definitions. You're not saying that I should generally use opacity instead of having different background colours for light and dark mode.

And if I'm creating new variables, I'm better off creating css variables rather than sass variables, because we're generally trying to minimise our dependency on sass since (a) it's a pain to compile everywhere and (b) css has become much more capable since sass was invested.  But it's still worthwhile to provide a way to set the default value of a css variable from a sass variable, since there's currently an easy way for the user to provide custom values for sass variables?

Reply:
That's spot on and well summarized 😄